### PR TITLE
Add new prismic features (dates + singleton)

### PR DESCRIPTION
**Pull request by arnaudlewis**
- add first and last publication date
- query singleton doc

### CODE REVIEW

**Comment by erwan** [comment]
Maybe camelCase to be consistent with the rest of the kit?

File: lib/api.js [hunk 1]
```diff
@@ -518,6 +518,8 @@ var parseDoc = function(json) {
     json.href,
     json.tags,
     slugs,
+    jsogit.first_publication_date,
```
Comment (martijndeh) [inline lib/api.js]: Typo.

File: lib/api.js [hunk 2]
```diff
@@ -518,6 +518,8 @@ var parseDoc = function(json) {
...
+    json.last_publication_date,
```
Comment (martijndeh) [inline lib/api.js]: Can we wrap these with a `new Date(..)` to actually cast them to a date? Else they are a string and everyone in user land needs to convert them.

File: lib/documents.js [hunk 3]
```diff
@@ -506,7 +506,7 @@ WithFragments.prototype = {
  * @global
  * @alias Doc
  */
-function Document(id, uid, type, href, tags, slugs, data) {
+function Document(id, uid, type, href, tags, slugs, firstPublicationDate, lastPublicationDate, data) {
```
Comment (martijndeh) [inline lib/documents.js]: Is camel case really the casing the library is using? I see more train case than camel case see `form_method` in `Form` constructor, the arguments in `Response` constructor etc.

File: lib/documents.js [hunk 4]
```diff
@@ -547,6 +547,14 @@ function Document(id, uid, type, href, tags, slugs, data) {
    */
   this.data = data;
   /**
+   * The first publication date of the document
+   */
+  this.firstPublicationDate = new Date(firstPublicationDate);
```
Comment (martijndeh) [inline lib/documents.js]: Can these dates be null? I think so, right?

If so, passing null to `new Date(..)` will return a date with year 1970. Not too sure if we want to add some extra check?

File: lib/documents.js [hunk 5]
```diff
@@ -547,6 +547,14 @@ function Document(id, uid, type, href, tags, slugs, data) {
    */
   this.data = data;
   /**
+   * The first publication date of the document
```
Comment (martijndeh) [inline lib/documents.js]: Missing `@type` annotation.

**Comment by erwan** [comment]
@arnaudlewis I just fixed the tests, could you rebase?

**Pull request merged**